### DISCUSSION TITLE
How to Download JSON for a Specific Language Filtered by Labels?

### DISCUSSION THREAD

**Discussion by MSL-BIT** [WeblateOrg/weblate#13082] [Q&A]
Context: We are using Weblate to manage translations in our project. While we can download the full JSON file for a specific language via the REST API, we need to filter it based on specific labels, such as static or dynamic.

What We've Tried: We used the following curl command to attempt filtering by labels:

```
curl -X GET \
     -H "Authorization: Token myToken" \
     "https://hosted.weblate.org/api/translations/<project>/<component>/<language>/file/?format=json&labels=static"
```

However, this returns all keys instead of only the ones tagged with the label static.

Questions:

Is it possible to filter the downloaded JSON file by specific labels (e.g., static or dynamic) directly using the Weblate REST API?
If not, is there an alternative way to achieve this, such as combining the /units/ API endpoint with additional processing to generate the desired JSON format?
Desired Outcome:

A JSON file containing only the translation keys for a specific language and label.
Support for excluding certain labels if needed.
Additional Details:

We are working on the <project> and <component> in Weblate.
Labels like static and dynamic are properly assigned to the keys in the Weblate UI.
Any guidance or suggestions would be highly appreciated. Thank you!

**Comment by nijel** [comment]
Yes, it is possible, just follow the documentation: https://docs.weblate.org/en/latest/api.html#get--api-translations-(string-project)-(string-component)-(string-language)-file-
